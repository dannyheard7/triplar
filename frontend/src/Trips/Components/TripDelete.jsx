import React from "react";
import api from "Trips/utils/trips.api.js";

import Modal from 'react-modal';

export default class TripDelete extends React.Component {
    customStyles = {
      content : {
        top                   : '40%',
        left                  : '40%',
        right                 : '40%',
        bottom                : '40%',
      }
    };

    constructor(props) {
        super(props);

        this.onDelete = this.onDelete.bind(this);
        this.closeModal = this.closeModal.bind(this);
    }

    closeModal() {
        this.props.history.push(`/trips/${this.props.trip.id}`)
    }

    onDelete(e) {
        api.deleteTrip(this.props.trip.id).then(({data}) => {
            if (data.data.deleteTrip.result === true) {
                this.props.history.push('/trips')
            }
        });
    };

    render() {
        return (
            <Modal isOpen={true} onRequestClose={this.closeModal} contentLabel={this.props.trip.name} style={this.customStyles}
                role="dialog">
                <h2>
                    Delete {this.props.trip.name}
                </h2>

                <p>
                    Are you sure you want to delete the trip '{this.props.trip.name}'?
                </p>

                <button onClick={this.closeModal}>Close</button>
                <button className="btn-danger" onClick={this.onDelete}>Delete Trip</button>

            </Modal>
        );
    }
}