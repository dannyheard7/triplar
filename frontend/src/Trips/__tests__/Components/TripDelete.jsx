import React from "react";

import TripDelete from "Trips/Components/TripDelete";
import ShallowRenderer from "react-test-renderer/shallow";
import {shallow} from "enzyme/build/index";

const renderer = new ShallowRenderer();

jest.mock('Trips/utils/trips.api.js');

describe('<TripDelete />', () => {
    const trip = {'id': 1, 'name': "Trip Test", 'created_by': "email@example.com"};
    const event = {
        target: {
            value: "",
            name: "",
        },
    };

    const props = {
        trip: trip,
        history: {push: jest.fn()}
    };


    test('<TripDelete /> renders correctly', () => {
        const result = renderer.render(<TripDelete  {...props} />);
        expect(result).toMatchSnapshot();
    });

    test('redirects to trips page on successful deletion', async () => {
        const container = shallow(<TripDelete {...props} />);

        container.instance().onDelete(event);
        await Promise.resolve();
        container.update();

        expect(props.history.push).toBeCalledWith('/trips');
    });
});