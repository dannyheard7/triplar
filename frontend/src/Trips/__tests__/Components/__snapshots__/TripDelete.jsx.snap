// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TripDelete /> <TripDelete /> renders correctly 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  closeTimeoutMS={0}
  contentLabel="Trip Test"
  isOpen={true}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
  style={
    Object {
      "content": Object {
        "bottom": "40%",
        "left": "40%",
        "right": "40%",
        "top": "40%",
      },
    }
  }
>
  <h2>
    Delete 
    Trip Test
  </h2>
  <p>
    Are you sure you want to delete the trip '
    Trip Test
    '?
  </p>
  <button
    onClick={[Function]}
  >
    Close
  </button>
  <button
    className="btn-danger"
    onClick={[Function]}
  >
    Delete Trip
  </button>
</Modal>
`;
